FWCore/Framework/src/exceptionContext.cc

```diff
@@ -0,0 +1,30 @@
+#include "FWCore/ServiceRegistry/interface/ESModuleCallingContext.h"
+#include "FWCore/ServiceRegistry/interface/ParentContext.h"
+#include "FWCore/ServiceRegistry/interface/ModuleCallingContext.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "FWCore/Framework/interface/ComponentDescription.h"
+
+#include <ostream>
+
+namespace edm {
+
+  void exceptionContext(cms::Exception& ex, ESModuleCallingContext const& mcc) {
```

Comment:
I find it a bit confusing that the function is declared in `Callback.h` but defined in `exceptionContext.cc`.

Reply:
It is only ever called in Callback.h and Callback.cc doesn't exist.

Reply:
I couldn't put it with the `ESModuleCallingContext` since it needs a header from `FWCore/Framework` which breaks our dependency layering

Reply:
Fair enough.